intelmq/bots/outputs/mcafee/output_esm_ip.py

```diff
@@ -0,0 +1,59 @@
+# -*- coding: utf-8 -*-
+"""
+
+ESMOutputBot connects to McAfee Enterprise Security Manager, and updates IP based watchlists
+
+Parameters:
+esm_ip: IP Address of ESM
+esm_user: username to connect to ESM
+esm_pw: Password of esm_user
+esm_watchlist: Destination watchlist to update
+field: field from intelMQ message to extract (e.g. destination.ip)
+
+"""
+
+from intelmq.lib.bot import Bot
+
+import json
+
+try:
+    from mfe_saw.esm import ESM
+except ImportError:
+    ESM = None
+
+
+class ESMIPOutputBot(Bot):
+
+    def init(self):
+        if ESM is None:
+            raise ValueError('Could not import mfe_saw. Please install it.')
+
+        self.esm = ESM()
+        try:
+            self.esm.login(self.parameters.esm_ip, self.parameters.esm_user, self.parameters.esm_pw)
+        except:
+            self.logger.error('Could not Login to ESM.')
+            self.stop()
+
+        watchlist_filter = {'filters': [{'name': 'IPAddress', 'id': 0}]}
+        try:
+            retVal = self.esm.post('sysGetWatchlists?hidden=false&dynamic=false&writeOnly=false&indexedOnly=false',
+                                   watchlist_filter)
+            for WL in retVal:
+                if (WL['name'] == self.parameters.esm_watchlist):
+                    watchlist = {'watchlist': {'value': WL['id']}, 'values': '["' + event.get(self.parameters.field) + '"]'}
```

Comment:
I looks like you never tested this bot. `event` is not defined. The whole block is part of the wrong function.